File: backend/app/schemas/alerts.py
"""Proactive portfolio alert schemas — deterministic signals only."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

AlertSeverity = Literal["critical", "warning", "info"]
AlertKind = Literal[
    "watchlist",
    "overloaded_owners",
    "blocked",
    "red_rag",
]


class AlertItem(BaseModel):
    id: str
    kind: AlertKind
    severity: AlertSeverity
    title: str
    detail: str
    count: int
    href: str = Field(description="In-app path for drill-down")


class AlertsResponse(BaseModel):
    data_version: str
    count: int
    alerts: list[AlertItem]

File: frontend/src/lib/api/alerts.ts
import { apiGet } from "./client";

export type AlertSeverity = "critical" | "warning" | "info";
export type AlertKind =
  | "watchlist"
  | "overloaded_owners"
  | "blocked"
  | "red_rag";

export type AlertItem = {
  id: string;
  kind: AlertKind;
  severity: AlertSeverity;
  title: string;
  detail: string;
  count: number;
  href: string;
};

export type AlertsResponse = {
  data_version: string;
  count: number;
  alerts: AlertItem[];
};

export function fetchAlerts(): Promise<AlertsResponse> {
  return apiGet<AlertsResponse>("/api/alerts");
}
